plugin-server/tests/main/ingestion-queues/buffer.test.ts

```diff
@@ -0,0 +1,50 @@
+import Piscina from '@posthog/piscina'
+import { DateTime } from 'luxon'
+
+import { runBuffer } from '../../../src/main/ingestion-queues/buffer'
+import { runInstrumentedFunction } from '../../../src/main/utils'
+import { Hub } from '../../../src/types'
+import { DB } from '../../../src/utils/db/db'
+import { createHub } from '../../../src/utils/db/hub'
+import { resetTestDatabase } from '../../helpers/sql'
+
+// jest.mock('../../../src/utils')
+jest.mock('../../../src/main/utils')
+
+describe('Event buffer', () => {
```

Comment:
A lot of the nuanced functionality is untested here:
- Locking-related behavior
- Races
- clearBufferLocks

I think testing these should have been considered critical - that's the kind of thing I have been pushing for in the recent month or so.